### ISSUE TITLE
SearchParams support?

### ISSUE THREAD

**Issue by jdin** [Gladear/haunted-router#3]
Can you add `searchParams` support?

I could not find it yet so I had to read it from `window.location`. 

But it could be much nicer to pass it directly to router function, for example:
```javascript
const loadMainPage = ({id}, {filter}) => {
  return html``
}
```
or maybe even like this:
```javascript
const loadMainPage = ({id,filter}) => {
  return html``
}
```

So that for url `/myID?filter=myFilter` I can get `{myID}, {myFilter}` or `{myID, myFilter}` as the argument(s) in `loadMainPage` function.

Or any other way :)

**Comment by Gladear** [comment]
Hi :wave:

Thanks for taking interest in this project, it's really wonderful to know that some people are using the projects you work on :blush:

I've got some issues with this proposal.

First, the second parameter is already the state passed in parameter to the `navigateTo` and `replaceTo` functions, or as an attribute on the `router-link` element. This is really the biggest issue, because the state can be retrieved using `history.state`, it's just a bit more practical.

Second, I'm quite sure I see less and less websites using search params, and even less web apps (which are the target of routers in general, in my experience). For those who want to use them, parsing them is as simple as this one-liner:
```js
const searchParams = new URLSearchParams(location.search);
```
Moreover, putting them in an object would cost another operation, which, again, would be a waste for most people.
```js
// This would create an object everytime the user navigates, most of time for nothing
[...searchParams.entries()].reduce((obj, [key, value]) => ({ ...obj, [key]: value }), {})
```

Therefore, I'm not sure I want to add this functionnality for everyone.

I've got to possible solutions:
- Create a `useSearchParams` hook,
- Or _"decorate"_ the `useRoute` hook so that it's behavior is changed when it's required

What do you think about these ?

**Comment by jdin** [comment]
Hi @Gladear, I really like the idea of useSearchParams hook and I am gonna use it in my project.
Thanks for the hint!
But don't you think that `haunted-router` can also be a good place for it?

**Comment by Gladear** [comment]
Yes, definitely ! I was just wondering what was the best way to make search params easy to use 😊 I'll look into it when I have some time. Probably tomorrow or this week-end 😉

**Comment by Gladear** [comment]
The `useSearchParams` hook is available in v0.2.0 :smiley:
